src/mlpack/methods/ne/gene.hpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file gene.hpp
+ * @author Bang Liu
+ *
+ * Definition of the LinkGene and NeuronGene class.
+ */
+#ifndef MLPACK_METHODS_NE_GENE_HPP
+#define MLPACK_METHODS_NE_GENE_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This is enumeration of neuron types.
+ */
+enum NeuronType {
+  NONE = 0,
+  INPUT,
+  BIAS,
+  HIDDEN,
+  OUTPUT
+};
+
+/**
+ * This is enumeration of activation function types.
+ */
+enum ActivationFuncType {
+    SIGMOID = 0,
+    TANH,
+    LINEAR,
+    RELU,
+};
+
+/**
+ * This class defines a link gene.
+ */
+class LinkGene {
+ public:
+  // Default constructor.
+  LinkGene() {}
+
+  // Parametric constructor.
+  LinkGene(unsigned int fromNeuronId,
```

Comment:
Can you use ```size_t``` instead of ```unsigned int```. It's guaranteed to be big enough to hold the size of the biggest object on your system. Also, I would recommend to use ```const```here, so ```const size_t fromNeuronId```.